src/mlir/generator.cc

```diff
@@ -12,6 +12,64 @@
 
 namespace mlir::verona
 {
+  void MLIRGenerator::initializeArithmetic()
+  {
+    // FIXME: Using MLIR standard dialect comes with complications, since the
+    // arithmetic in MLIR is mostly for ML and doens't have the semantics we
+    // need for generic lowering. We should probably use the LLVM dialect + LLVM
+    // intrinsic calls for the rest.
+
+    // Floating-point arithmetic
+    arithmetic.emplace("std.absf", 1);
```

Comment:
Why not write this as an initialiser list in the constructor?  Something like:

```
arithmetic({ "std.absf", 1},
           { "std.ceilf", 1}, 
           ...)
```

Or, since this is a simple string-to-number map, remove this field entirely and just have a function that does the lookup and use `llvm::StringSwitch`, which typically generates much better code for this kind of thing.

Reply:
> Why not write this as an initialiser list in the constructor?

It grew from a few examples.

> Or, since this is a simple string-to-number map, remove this field entirely and just have a function that does the lookup and use `llvm::StringSwitch`, which typically generates much better code for this kind of thing.

I wasn't sure this wouldn't need multiple points of insertion. But I think we can cross that bridge when we get there.

I'll change to `StringSwitch`.